plugins/woocommerce/src/Admin/Notes/Notes.php

```diff
@@ -34,12 +34,22 @@ public static function init() {
 	 * @return array Array of arrays.
 	 */
 	public static function get_notes( $context = 'edit', $args = array() ) {
-		$data_store = self::load_data_store();
-		$raw_notes  = $data_store->get_notes( $args );
-		$notes      = array();
+		$data_store  = self::load_data_store();
+		$raw_notes   = $data_store->get_notes( $args );
+		$notes       = array();
+		$user_locale = get_user_locale();
 		foreach ( (array) $raw_notes as $raw_note ) {
 			try {
-				$note                               = new Note( $raw_note );
+				$note = new Note( $raw_note );
+				if ( $user_locale !== $note->get_locale() ) {
+					/**
+					 * Filter the note from db. This is used to modify the note before it is returned when the user's locale does not match the note's locale.
+					 *
+					 * @since 6.9.0
+					 * @param Note $note The note object from the database.
+					 */
+					$note = apply_filters( 'woocommerce_get_note_from_db', $note );
```

Comment:
@chihsuan I'm wondering if it's worth checking the locale first before running? If we conditionally call a hook, I think it can lead to less predictable behaviour and can open up the hook for more uses outside of locale

Reply:
Good question! 

I thought if we add this check, we don't need to check the locale in the individual filter functions and reduce non-necessary calls, but also agree with you that opening up the hook could make it more useful. 

I think I kind of changed my mind. We should give developers the ability to control this.

Reply:
I've removed the check in https://github.com/woocommerce/woocommerce/pull/34038/commits/54bc829938db121750b0d2f042ea26ac22ea3562.